change: [M3-10242] - Move LKE endpoints from v4 to v4beta endpoints to allow restricted billing user access

mjac0bs opened the pull request:
## Description 📝

We've made a decision as a team to switch over to the v4beta endpoints (a v4 superset) app-wide, but we will do this section by section.

This ticket will focus on addressing v4beta endpoints that are impacted by the LKE-E account capability check. Rather than toggle between v4beta and v4 depending on whether the LKE-E feature is enabled, we will use v4beta all the time.

This PR handles a lot of clean up and fixes the current issue where, even if a restricted billing user has read access to an existing LKE-E cluster, they still cannot see the features of an LKE-E cluster (chip, correct version, configure pool action) because they rely on checks of the /account endpoint capability to enable that part of the UI.

## Changes  🔄
- Change all kubernetes endpoints from `/v4` to `/v4beta` in `api-v4`; clean up duplicate endpoints
   - Note that node pool endpoints were already switched over in a previous PR
- Update kubernetes queries (remove unneeded logic/query keys and duplicated functions)
- Clean up logic for toggling between kubernetes `/v4` and `/v4beta` endpoints in components
   - Remove `useKubernetesBetaEndpoint` hook now that we're always using the beta endpoint
- Adjusts some conditional logic to stop the capability from gating the enterprise UI, instead relying on the `cluster.tier` of `enterprise`
   - This allows restricted billing users with read-only/read-write enterprise cluster access to see the correct UI (enterprise chip, versions, configure pool option).
- Adjusts the post-LA feature check to not check the account capability _if_ the user is restricted from billing access. Instead, just check the post-LA feature flag.
   - This should be okay for LKE-E, because the LA feature is already gated by the "Kubernetes Enterprise" , which controls ability for a cluster to return an "enterprise" `tier`.
   - For LKE standard, this change allows restricted billing users with the `add_lkes` grant to see the new "Configure Pool" UI, rather than the "Add" UI in the add node pool flows.

### Scope 🚢

 Upon production release, changes in this PR will be visible to:

- [x] All customers
- [ ] Some customers (e.g. in Beta or Limited Availability)
- [ ] No customers / Not applicable

## Target release date 🗓️

10/7

## How to test 🧪

### Prerequisites

(How to setup test environment)

- Have the LKE-Enterprise customer tag
- Ensure the LKE-E feature flags are on in your local environment (enabled, la, postLa, phase2Mtc flags)
- Have a normal restricted user account with No Billing Access
- Have an existing LKE-E cluster on the account (or create one as a non-restricted Billing user) that the restricted user has access to

### Reproduction steps

(How to reproduce the issue, if applicable)

- [ ] Log into production Cloud Manager as a restricted user with no billing access
- [ ] Observe that even though you have the LKE-Enterprise customer tag + feature enabled, for any existing LKE-E clusters, LKE-E features won't show up (e.g. the ENTERPRISE chip, correct version, configure pool)

### Verification steps

(How to verify changes)

- [ ] Check out this branch and run it locally, logging in as a **restricted user with no billing access**.
- [ ] Observe that for any existing LKE-E clusters, the cluster is designated as an LKE-E cluster (chip, correct version, ability to configure the node pool).
- [ ] Observe that for any standard LKE clusters, you can create an LKE cluster, the correct version displays, and you cannot configure the node pool.
- [ ] Remove the `lke-enterprise-eap` customer tag (or use the account mocks and remove `Kubernetes Enterprise`) and confirm no regressions.
   - [ ] For a standard tier cluster, the "Configure Pool" button/flow should be visible in the **Create Cluster > Add Node Pool flow** (but not the standard LKE cluster details page, see comments).
   - [ ] Without the customer tag, beyond the correct LKE-E cluster version, you shouldn't see anything else related to LKE-E, even when viewing an existing LKE-E cluster.
- [ ] Ensure all LKE related unit and integration tests pass.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All tests and CI checks are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

<!-- This content will not appear in the rendered Markdown 

## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**

- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

-->

mjac0bs commented on packages/manager/src/features/Kubernetes/ClusterList/ClusterChips.test.tsx:
Removed this test because we now render the chip for any cluster of the enterprise tier. (The original creation of the cluster did depend on the account capability/feature enablement for the customer.)

mjac0bs reviewed (commented):
Just did a quick self review.

dwiley-akamai reviewed (commented):
Verification steps look good to me except:

> Remove the `lke-enterprise-eap` customer tag --> standard tier cluster, the "Configure Pool" button/flow should be visible in the Add Node Pool flows.

I'm also not observing this 🤔

coliu-akamai commented:
ahh sorry for the confusion, I'd been looking at the LKE cluster's details page and Add Node Pool flow there: 
<img width="677" height="74" alt="image" src="https://github.com/user-attachments/assets/4c454986-4bc0-4789-bd30-14e5663b5a46" />

(I saw the Configure Pool action menu item for an LKE-E cluster in its detail page, so I assumed I should look there for regular clusters as well)
<img width="285" height="106" alt="image" src="https://github.com/user-attachments/assets/0c9e1e76-817f-450e-98b7-df80420a0caf" />

Confirming I can see the Configure Pool flow in the LKE create flow as per the video for a regular LKE cluster!

mjac0bs commented:
> (I saw the Configure Pool action menu item for an LKE-E cluster in its detail page, so I assumed I should look there for regular clusters as well)

@coliu-akamai  Ahh, I see. Sorry for the confusion, that's on me. "Configure Pool" is not an action menu option for regular clusters because at this point all that drawer contains are LKE-E specific options.

This differs from the create flow, where the user can configure their pool to set the pool size.

Ultimately, "Resize Pool" should probably be replaced by a resize option within "Configure Pool" that is available to all cluster tiers but... that's not happening right now. Or any time soon, given priorities. 😅

dwiley-akamai reviewed (approved):
I had also been looking for the "Configure Pool" button on the Cluster Detail page, but I do see it in the LKE Create flow under the described conditions for a restricted user. 👍🏾

mjac0bs commented:
Latest changes:

- Removed `useLkeStandardOrEnterpriseVersions` and entirely (and its test coverage)
- Updated the create spec tests that use the `mockGetKubernetesVersion` query
   - Once that function was no longer in use, removed it
- Removed `useKubernetesVersionQuery` and `getAllKubernetesVersions` from the kubernetes queries file since no longer in use in the codebase

...if all tests don't pass cleanly with these changes, I'll revert that last commit and put it in its own PR.
